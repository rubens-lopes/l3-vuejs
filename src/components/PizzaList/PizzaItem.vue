<template>
  <li class="font-light flex">
    <pizza-id>{{ pizza.id }}</pizza-id>

    <pizza-name>{{ pizza.name }}</pizza-name>

    <pizza-gluten>
      <font-awesome-icon
        :icon="getGlutenIcon({ isGlutenFree: pizza.isGlutenFree })"
      ></font-awesome-icon>
    </pizza-gluten>
  </li>
</template>

<script lang="ts">
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import PizzaId from "@/components/PizzaList/PizzaId.vue";
import PizzaName from "@/components/PizzaList/PizzaName.vue";
import PizzaGluten from "@/components/PizzaList/PizzaGluten.vue";

export default {
  components: {
    FontAwesomeIcon,
    PizzaId,
    PizzaName,
    PizzaGluten,
  },
  props: {
    // eslint-disable-next-line no-undef
    pizza: Object as () => Pizza,
  },
  methods: {
    getGlutenIcon({ isGlutenFree = false } = {}) {
      return isGlutenFree ? `check` : `times`;
    },
  },
};
</script>
