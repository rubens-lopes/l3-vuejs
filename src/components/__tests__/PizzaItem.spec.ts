import { shallowMount } from "@vue/test-utils";
import PizzaItem from "@/components/PizzaList/PizzaItem.vue";

describe("PizzaItem.vue", () => {
  it("renders a pizza properly", () => {
    const pizza = {
      id: 2,
      name: "Moda da casa",
      isGlutenFree: false,
    };

    const wrapper = shallowMount(PizzaItem, {
      propsData: { pizza },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
