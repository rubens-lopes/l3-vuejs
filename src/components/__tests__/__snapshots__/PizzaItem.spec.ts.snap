// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaItem.vue renders a pizza properly 1`] = `
<li class="font-light flex">
  <pizza-id-stub>2</pizza-id-stub>
  <pizza-name-stub>Moda da casa</pizza-name-stub>
  <pizza-gluten-stub>
    <font-awesome-icon-stub icon="times"></font-awesome-icon-stub>
  </pizza-gluten-stub>
</li>
`;
